evidence/pool.go

```diff
@@ -218,7 +270,55 @@ func (evpool *Pool) ValidatorLastHeight(address []byte) int64 {
 	return h
 }
 
-func (evpool *Pool) removeEvidence(
+// State returns the current state of the evpool.
+func (evpool *Pool) State() sm.State {
+	evpool.mtx.Lock()
+	defer evpool.mtx.Unlock()
+	return evpool.state
+}
+
+func (evpool *Pool) addPendingEvidence(evidence types.Evidence) error {
+	evBytes := cdc.MustMarshalBinaryBare(evidence)
+	key := keyPending(evidence)
+	return evpool.evidenceStore.Set(key, evBytes)
+}
+
+func (evpool *Pool) removePendingEvidence(evidence types.Evidence) {
+	key := keyPending(evidence)
+	if err := evpool.evidenceStore.Delete(key); err != nil {
+		evpool.logger.Error("Unable to delete pending evidence", "err", err)
+	}
+}
+
+// listEvidence lists up to maxNum pieces of evidence for the given prefix key.
+// It is wrapped by PriorityEvidence and PendingEvidence for convenience.
+// If maxNum is -1, there's no cap on the size of returned evidence.
+func (evpool *Pool) listEvidence(prefixKey byte, maxNum int64) (evidence []types.Evidence, error error) {
```

Comment:
sure - I will change that

Reply:
I think this is int64 because everything else is int64

Reply:
do you just want me to convert it?

Reply:
I've left it as an int - let me know if you want me to change it and if so if you want me to change the others as well

Reply:
what do you mean by everything else? it may be fine to leave as is. I just don't understand why it has to be int64

Reply:
well maxNum is int64 because pending evidence requires int64 and that's because when we calculate the amount of evidence it's the division of two int64's and so on